coderd/prometheusmetrics/collector.go

```diff
@@ -0,0 +1,80 @@
+package prometheusmetrics
+
+import (
+	"sync"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type CachedGaugeVec struct {
+	m sync.Mutex
+
+	gaugeVec *prometheus.GaugeVec
+	records  []vectorRecord
+}
```

Comment:
Can you doc the usage? And the why?

Eg:

> CachedGaugeVec does .....
Calling WithLabelValues will update the internal gauge value. The value will not be returned by 'Collect' until 'Commit' is called. 
'Commit' will reset the internal value, requiring the next set of values to build upon a completely reset metric.

Or something...

Reply:
Added 👍